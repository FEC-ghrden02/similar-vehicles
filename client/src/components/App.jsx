import React from 'react';
import SimilarVehicles from './SimilarVehicles.jsx';


class App extends React.Component{
    constructor(props) {
        super(props); 
        this.state = {
            similarVehicles: [], 
            hover: false
        }; 

        this.onMouseOut = this.onMouseOut.bind(this); 
        this.onMouseOver = this.onMouseOver.bind(this); 
        this.getSimilarVehicles = this.getSimilarVehicles.bind(this); 
    } 

    componentDidMount() {
        this.getSimilarVehicles()
    } 

    getSimilarVehicles () {
        fetch('http://localhost:3000/api/similar_vehicles')
        .then((results) => {
           return results.json()
        })
        .then((data) => {
            this.setState({
                similarVehicles: data.results
            })
        })
        .catch((err) => {
            if (err) {
                console.log("Ecountered Error in getSimilarVehicles", err)
            }
        })

    }; 

    onMouseOver() {
        
        this.setState(
            {
                hover: true
            }
        )
    }; 

    onMouseOut() {
        
        this.setState(
            {
                hover: false
            }
        )
    }; 

    render() {
        return(
            <div className="similarVehiclesContainer">
                <SimilarVehicles 
                similarVehicles={this.state.similarVehicles} 
                hover={this.state.hover} 
                onMouseOver={this.onMouseOver} 
                onMouseOut={this.onMouseOut} /> 
            </div>
        ); 
    } 
} 

export default App;