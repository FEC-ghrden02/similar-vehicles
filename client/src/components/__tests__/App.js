import React from 'react';
import App from '../App.jsx';
import { shallow } from 'enzyme';

describe('App', () => {
  let wrapper;
  window.fetch = jest.fn();

  beforeEach(() => {
    wrapper = shallow(<App />);
  });

  it('should match snapshoot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  // expect(App.state).toMatchObject({
  //   similarVehicles: []
  // })
});