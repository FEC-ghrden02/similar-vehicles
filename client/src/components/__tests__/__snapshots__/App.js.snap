// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match snapshoot 1`] = `undefined`;
